### PR TITLE
Standardizes mate info computation in SAMRecordSetBuilder.

### PR DESCRIPTION
**Pull request by nh13**
<edited>

@yfarjoun or @tfenne care to take a look?

### CODE REVIEW

File: src/main/java/htsjdk/samtools/SAMRecordSetBuilder.java [hunk 1]
```diff
@@ -358,13 +358,8 @@ public void addPair(final String name, final int contig, final int start1, final
         end1.setMappingQuality(255);
         end1.setReadPairedFlag(true);
         end1.setProperPairFlag(true);
-        end1.setMateReferenceIndex(contig);
-        end1.setAttribute(SAMTag.MC.name(), readLength + "M");
-        end1.setMateAlignmentStart(start2);
-        end1.setMateNegativeStrandFlag(true);
         end1.setFirstOfPairFlag(end1IsFirstOfPair);
         end1.setSecondOfPairFlag(!end1IsFirstOfPair);
-        end1.setInferredInsertSize((int) CoordMath.getLength(start1, CoordMath.getEnd(start2, this.readLength)));
```
Comment (yfarjoun) [inline src/main/java/htsjdk/samtools/SAMRecordSetBuilder.java]: setinferredinsertsize is reset to zero in setMateInfo, which is different from 
```
CoordMath.getLength(start1, CoordMath.getEnd(start2, this.readLength)));
```

File: src/main/java/htsjdk/samtools/SAMRecordSetBuilder.java [hunk 2]
```diff
@@ -383,13 +378,8 @@ public void addPair(final String name, final int contig, final int start1, final
         end2.setMappingQuality(255);
         end2.setReadPairedFlag(true);
         end2.setProperPairFlag(true);
-        end2.setMateReferenceIndex(contig);
-        end2.setAttribute(SAMTag.MC.name(), readLength + "M");
```
Comment (yfarjoun) [inline src/main/java/htsjdk/samtools/SAMRecordSetBuilder.java]: this differs in setMateInfo (MC)

File: src/main/java/htsjdk/samtools/SAMRecordSetBuilder.java [hunk 3]
```diff
@@ -383,13 +378,8 @@ public void addPair(final String name, final int contig, final int start1, final
...
-        end2.setMateAlignmentStart(start1);
-        end2.setMateNegativeStrandFlag(false);
         end2.setFirstOfPairFlag(!end1IsFirstOfPair);
         end2.setSecondOfPairFlag(end1IsFirstOfPair);
-        end2.setInferredInsertSize(end1.getInferredInsertSize());
```
Comment (yfarjoun) [inline src/main/java/htsjdk/samtools/SAMRecordSetBuilder.java]: this is odd...should be negative, no? the original seems wrong.

File: src/main/java/htsjdk/samtools/SAMRecordSetBuilder.java [hunk 4]
```diff
@@ -358,13 +358,8 @@ public void addPair(final String name, final int contig, final int start1, final
         end1.setMappingQuality(255);
         end1.setReadPairedFlag(true);
         end1.setProperPairFlag(true);
-        end1.setMateReferenceIndex(contig);
-        end1.setAttribute(SAMTag.MC.name(), readLength + "M");
```
Comment (yfarjoun) [inline src/main/java/htsjdk/samtools/SAMRecordSetBuilder.java]: here too (MC)

**Comment by yfarjoun** [comment]
back to you @nh13

**Comment by yfarjoun** [comment]
friendly ping @nh13 . this is in your court. ⚽️

**Comment by nh13** [comment]
@yfarjoun thanks, assigned it to myself for visibility.

**Comment by yfarjoun** [comment]
📳

**Comment by nh13** [comment]
I hear dead PRs. I'll get to it 👍

**Comment by yfarjoun** [comment]
Thanks!

File: src/main/java/htsjdk/samtools/SAMRecordSetBuilder.java (hunk 1 above)
Comment (nh13) [inline src/main/java/htsjdk/samtools/SAMRecordSetBuilder.java]: The call to setMateInfo on line 398 does the right thing: https://github.com/samtools/htsjdk/blob/master/src/main/java/htsjdk/samtools/SamPairUtil.java#L271-L273

Maybe I am misunderstanding your comment?

File: src/main/java/htsjdk/samtools/SAMRecordSetBuilder.java (hunk 2 above)
Comment (nh13) [inline src/main/java/htsjdk/samtools/SAMRecordSetBuilder.java]: Not for this method.

File: src/main/java/htsjdk/samtools/SAMRecordSetBuilder.java (hunk 3 above)
Comment (nh13) [inline src/main/java/htsjdk/samtools/SAMRecordSetBuilder.java]: Yup.

**Comment by nh13** [comment]
@yfarjoun I don't understand your comments, perhaps because I haven't looked at this in a long time!  The main goal was to stop setting the mate info directly and use the calculation in `SamPairUtil.setMateInfo`, which is the "standardized" way in htsjdk to set mate info given you have set the info for two records individually.  What am I missing?

File: src/main/java/htsjdk/samtools/SAMRecordSetBuilder.java (hunk 3 above)
Comment (yfarjoun) [inline src/main/java/htsjdk/samtools/SAMRecordSetBuilder.java]: OK. I get it now. thanks!

**Comment by yfarjoun** [comment]
@nh13 this can be merged!

**Pull request merged**